File: web/utils/JinjaHelper.py
from urllib import urlencode

__author__ = 'stevelowenthal'


def makeURL(url, *args):
    url_params = [(args[i],args[i+1].encode("UTF-8")) for i in range(0,len(args),2)]
    return url + '?' + urlencode(url_params)

File: web-java/src/main/java/retail/helpers/jinjahelper/JinjaHelper.java
package retail.helpers.jinjahelper;

import java.io.UnsupportedEncodingException;
import java.net.URLEncoder;

/**
 * DataStax Academy Sample Application
 * <p/>
 * Copyright 2015 DataStax
 */
public class JinjaHelper {
    // The elements are in pairs ("q","\"title:usb\"","fq","somepredicate")
    public static String makeURL(String url, String ... elements) throws UnsupportedEncodingException {

        StringBuilder s = new StringBuilder(url);

        char separator = '?';
        for (int i = 0; i < elements.length; i += 2) {
            s.append(separator)
                    .append(elements[i])
                    .append('=')
                    .append(URLEncoder.encode(elements[i+1],"UTF-8"));
           separator = '&';
        }
        return s.toString();
    }

}
